### PR TITLE
feat: add error handling for `generate-all.sh`

### PR DESCRIPTION
**Pull request by diegomarquezp**
With this PR, if one of the libraries generated by `generate-one.sh` fails, the now captured std* output will be stored in a file under `failed-library-generations` named the same as the client library (i.e. one file will be generated with std* of each failed lib). `geneate-all.sh` continues normally after the failed output is saved.

### CODE REVIEW

File: spring-cloud-generator/generate-all.sh
Comment (emmileaf) [inline spring-cloud-generator/generate-all.sh:46]: I think this rm is redundant (and fails) because you already use `mv tmp-generate-one-output` instead of `cp tmp-generate-one-output` above
```suggestion
```

File: spring-cloud-generator/generate-all.sh
Comment (emmileaf) [inline spring-cloud-generator/generate-all.sh:21]: Nit: perhaps we could rename this folder to something like `preview-module-errors`?

**Review by emmileaf — APPROVED** [review]
Thank you - this change is super helpful for troubleshooting!

File: spring-cloud-generator/generate-all.sh
Comment (zhumin8) [reply spring-cloud-generator/generate-all.sh:21]: I think either name is fine, or something along the lines of `generation-error-logs`, as long as we don't intend to merge in the changes to main. It would be weird to have a failed/errors in our repo's main.
I imagine the process goes something like this:
- trigger generation from GithubAction
- Check on generated PR, look for anything in this folder. 
- In case there are errors, we probably don't want to merge in the pr. Will try to troubleshot or exclude and regenerate.
- If no errors occurred, this folder will not be populated to the PR.

**Comment by emmileaf** [comment]
@diegomarquezp bringing in the latest changes from main should unblock the releaseCheck issue here - thanks!

File: spring-cloud-generator/generate-all.sh
Comment (diegomarquezp) [reply spring-cloud-generator/generate-all.sh:21]: The action checks if any files are in `preview-module-errors`

Comment (diegomarquezp) [reply spring-cloud-generator/generate-all.sh:14]: I will modify the folder name to a more understandable one once #1341 is merged

**Pull request merged**
